sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -855,6 +855,11 @@ object TypeCoercion {
       case Divide(l @ CalendarIntervalType(), r @ NumericType()) =>
         DivideInterval(l, r)
 
+      case b @ BinaryOperator(l @ CalendarIntervalType(), r @ NullType()) =>
```

Comment:
This is a little hacky. Maybe we should introduce `UnresolvedMultiply` and `UnresolvedDivide`, so that we don't need to hack the type coercion rules.

We can try it in followup.